In [8]:
import ssl
import seaborn as sns

ssl._create_default_https_context = ssl._create_unverified_context
df = sns.load_dataset("titanic")
print(df.head())

   survived  pclass     sex   age  sibsp  parch     fare embarked  class  \
0         0       3    male  22.0      1      0   7.2500        S  Third   
1         1       1  female  38.0      1      0  71.2833        C  First   
2         1       3  female  26.0      0      0   7.9250        S  Third   
3         1       1  female  35.0      1      0  53.1000        S  First   
4         0       3    male  35.0      0      0   8.0500        S  Third   

     who  adult_male deck  embark_town alive  alone  
0    man        True  NaN  Southampton    no  False  
1  woman       False    C    Cherbourg   yes  False  
2  woman       False  NaN  Southampton   yes   True  
3  woman       False    C  Southampton   yes  False  
4    man        True  NaN  Southampton    no   True  


In [9]:
print(df.shape)

(891, 15)


In [10]:
print(df.dtypes)

survived          int64
pclass            int64
sex                 str
age             float64
sibsp             int64
parch             int64
fare            float64
embarked            str
class          category
who                 str
adult_male         bool
deck           category
embark_town         str
alive               str
alone              bool
dtype: object


Types redondants ou mal choisis :
- survived devrait etre un bool
- age devrait être int plutôt que float
- sex et who est redondant
- Alive pourrait être un bool

In [11]:
df.isna().sum()

survived         0
pclass           0
sex              0
age            177
sibsp            0
parch            0
fare             0
embarked         2
class            0
who              0
adult_male       0
deck           688
embark_town      2
alive            0
alone            0
dtype: int64

In [14]:
df["class"].value_counts()

class
Third     491
First     216
Second    184
Name: count, dtype: int64

In [15]:
df["embark_town"].value_counts()

embark_town
Southampton    644
Cherbourg      168
Queenstown      77
Name: count, dtype: int64

Constats pour la suite:
- il faudra changer le type de certaines colonne et supprimer les redondantes
- la colonne deck présente beaucoup de valeurs manquantes (688 sur 891), il faudra sans passer pour la suite au risque de fausser certains résultats
- on remarque que la majorité des passagers on embarquer à Southampton